index.d.ts

```diff
@@ -0,0 +1,41 @@
+import { Model, ModelClass, Page } from 'objection-2'
+import { Ability } from '@casl/ability'
+import { AccessControl } from 'role-acl-3'
+
+type AuthorizeOptions = {
+  defaultRole?: string
+  unauthenticatedErrorCode?: number
+  unauthorizedErrorCode?: number
+  userFromResult?: boolean
+  contextKey?: string
+  roleFromUser?: (user: Model) => string
+  resourceAugments?: { true: boolean; false: boolean; undefined: undefined }
+}
+
+declare module 'objection-authorize' {
+  class AuthQueryBuilder<M extends Model, R = M[]> extends Model.QueryBuilder<
+    M,
+    R
+  > {
+    ArrayQueryBuilderType: AuthQueryBuilder<M, M[]>
+    SingleQueryBuilderType: AuthQueryBuilder<M, M>
+    NumberQueryBuilderType: AuthQueryBuilder<M, number>
+    PageQueryBuilderType: AuthQueryBuilder<M, Page<M>>
+  }
+  class AuthorizeModelClass extends Model {
+    QueryBuilderType: AuthQueryBuilder<this>
+    QueryBuilder: AuthQueryBuilder<this>
+  }
+
+  export default function authorize (
```

Comment:
So... unfortunately this suffers from the same issue pointed out in https://github.com/JaneJeon/objection-hashid/pull/51#discussion_r373315460, where:
1. the type information is not actually usable from the outside world, and
2. custom query methods (specifically, `.authorize()` and `.action()`) are erased.

:/